Src/VsVimShared/Implementation/Settings/VimCollectionSettingsStore.cs

```diff
@@ -0,0 +1,131 @@
+        }
+
+        private T GetProperty<T>(string name, T defaultValue, Func<string, string, T, T> getValueFunc)
+        {
+            EnsureCollectionExists();
+            if (_propertyMap.TryGetValue(name, out object obj))
+            {
+                if (obj is T t)
+                {
+                    return t;
+                }
+
+                // Property exists at a different type. Fall back to the backing settings store
+                _propertyMap.Remove(name);
+            }
+
+            try
+            {
+                var value = getValueFunc(_collectionPath, name, defaultValue);
+                _propertyMap[name] = value;
+                return value;
+            }
+            catch (Exception e)
+            {
+                Report($"Unable to retrieve value {name} from the settings store", e);
+                return defaultValue;
+            }
+        }
+
+        private void SetProperty<T>(string name, T value, Action<string, string, T> setValueFunc)
+        {
+            EnsureCollectionExists();
+            _propertyMap.Remove(name);
+            try
+            {
+                setValueFunc(_collectionPath, name, value);
+                _propertyMap[name] = value;
+            }
+            catch (Exception e)
+            {
+                Report($"Unable to set value {name} to the settings store", e);
+            }
+        }
+
+        internal bool GetBoolean(string propertyName, bool defaultValue) =>
+            GetProperty<bool>(propertyName, defaultValue, _getBoolFunc);
+
+        internal void SetBoolean(string propertyName, bool value) =>
+            SetProperty(propertyName, value, _setBoolFunc);
+
+        internal string GetString(string propertyName, string defaultValue) =>
+            GetProperty<string>(propertyName, defaultValue, _getStringFunc);
+
+        internal void SetString(string propertyName, string value) =>
+            SetProperty(propertyName, value, _setStringFunc);
+
+        internal int Getint(string propertyName, int defaultValue) =>
+            GetProperty<int>(propertyName, defaultValue, _getIntFunc);
+
+        internal void Setint(string propertyName, int value) =>
+            SetProperty(propertyName, value, _setIntFunc);
+
+        private void EnsureCollectionExists()
+        {
+            if (!_hasCreatedCollection)
+            {
+                try
+                {
+                    if (!_settingsStore.CollectionExists(_collectionPath))
+                    {
+                        _settingsStore.CreateCollection(_collectionPath);
+                    }
+
+                    _hasCreatedCollection = true;
+                }
+                catch (Exception e)
+                {
+                    Report("Unable to create the settings collection", e);
+                }
+            }
+        }
+
+        private void Report(string message, Exception e)
+        {
+            message = message + ": " + e.Message;

```

Comment:
I think an interpolated string is both more efficient and more readable, so it's a win/win.

Reply:
Good point. Let me fix that.